src/App.tsx

```diff
@@ -8,52 +8,47 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
+import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as authorsActions } from './features/authors';
 import { User } from './types/User';
+import { clear, postsAsync } from './features/posts';
+import { actions } from './features/selectedPost';
 import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const { posts, loaded, hasError } = useAppSelector(state => state.posts);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const author = useAppSelector(state => state.authors);
+  const dispatch = useAppDispatch();
+  const selectedPost = useAppSelector(state => state.selectedPost);
```

Comment:
Recomendation: Create selector and use them
`const selectedPostSelector = (state: State) => state.selectedPost;`
You can reused it everywhere, and can be sure if something will change - you will change in one place - so everything will works

```
const selectedPost = useAppSelector(selectedPostSelector);
```